lib/segment/src/json_path/string.rs

```diff
@@ -0,0 +1,119 @@
+use std::borrow::Cow;
+use std::fmt::{Display, Formatter};
+use std::str::FromStr;
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use validator::ValidationError;
+
+use super::JsonPathInterface;
+use crate::common::anonymize::Anonymize;
+use crate::common::utils::MultiValue;
+
+#[derive(Debug, PartialEq, Clone, Hash, Eq, Serialize, Deserialize, JsonSchema)]
+#[serde(transparent)]
+pub struct JsonPathString(pub String);
+
+impl Display for JsonPathString {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&self.0)
+    }
+}
+
+impl JsonPathInterface for JsonPathString {
+    fn value_get<'a>(&self, json_map: &'a serde_json::Map<String, Value>) -> MultiValue<&'a Value> {
+        #[allow(deprecated)]
+        crate::common::utils::get_value_from_json_map(&self.0, json_map)
+    }
+    fn value_set<'a>(
```

Comment:
Probably a nitpick, but please leave an empty line between functions or modules:

```suggestion
    }
    
    fn value_set<'a>(
```